packages/babel-types/src/definitions/core.js

```diff
@@ -704,7 +704,7 @@ defineType("WithStatement", {
       object: assertNodeType("Expression")
     },
     body: {
-      validate: assertNodeType("BlockStatement")
+      validate: assertNodeType("BlockStatement", "Statement")
```

Comment:
Yeah https://github.com/babel/babel/blob/master/packages/babel-types/src/definitions/core.js#L84-L97

```js

defineType("BlockStatement", {
  builder: ["body", "directives"],
  visitor: ["directives", "body"],
  fields: {
    directives: {
      validate: chain(assertValueType("array"), assertEach(assertNodeType("Directive"))),
      default: []
    },
    body: {
      validate: chain(assertValueType("array"), assertEach(assertNodeType("Statement")))
    }
  },
  aliases: ["Scopable", "BlockParent", "Block", "Statement"]
});
```

And `IfStatement` which does the same thing https://github.com/babel/babel/blob/master/packages/babel-types/src/definitions/core.js#L315-L330